src/gpu/nvidia/cudnn_convolution_impl.hpp

```diff
@@ -594,9 +659,21 @@ struct cudnn_convolution_impl_fwd_t : public cudnn_convolution_impl_base_t {
                 = utils::downcast<sycl_cuda_stream_t *>(service_stream);
         auto handle = cuda_stream->get_cudnn_handle();
 
-        CHECK(CUDNN_EXECUTE_FUNC_S(cudnnGetConvolutionForwardWorkspaceSize,
-                handle, descs[x], weights_desc, conv_desc, descs[y],
-                fwd_alg_kind, &scratchpad_size));
+        // The scratchpad size will need to be modified in
+        // cases where the dst_scaling is used and the output
+        // uses s8 values.
+        if (use_scales_dst_) {
```

Comment:
As I mentioned before there is already a scratchpad memory reserved which is called `post_op_scratch`. How does it work with the new scratchpad? I think the right way is to reuse it, because its main purpose was to handle cases with post ops and s8 dst:
https://github.com/oneapi-src/oneDNN/blob/274ee103c0c45dbb8aa255203d3d8627dda7ed3f/src/gpu/nvidia/cudnn_convolution.hpp#L101-L105
The current approach is:
`CONV(src, wei) -> dst(s32/f32) -> REORDER-> post_op_scratch(f32) -> POSTOPS -> REORDER->dst(dst_dt) -> DST_SCALE(dst_scale_f32) -> dst(dst_dt)`
The issue in the current approach is that dst scaling happens after quantization reorder, but it should move before:
`CONV(src, wei) -> dst(s32/f32) -> REORDER-> post_op_scratch(f32) -> POSTOPS  -> DST_SCALE(dst_scale_f32) -> REORDER->dst(dst_dt)`

Reply:
The current approach is doing the reorder after the dst scaling. That part is done during the `cudnnOpTensor``function call. That function scales de dst first and then applies a `reorder`.

Thanks for the feedback provided.